### PR TITLE
refactor: deprecate keepLifecycleScripts in favor of keepPackageScripts

### PR DESCRIPTION
**Pull request by SchnWalter**
## I'm submitting a...

```
[ ] Bug Fix
[ ] Feature
[x] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added

## Description

Deprecated the `keepLifecycleScripts` flag in favor of a better named flag: `keepPackageScripts`.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/lib/ng-package/entry-point/write-package.transform.ts [hunk 1]
```diff
@@ -138,19 +138,23 @@ async function writePackageJson(
 
   // Removes scripts from package.json after build
   if (packageJson.scripts) {
-    if (pkg.keepLifecycleScripts !== true) {
-      log.info(`Removing scripts section in package.json as it's considered a potential security vulnerability.`);
-      delete packageJson.scripts;
+    if (pkg.keepPackageScripts || pkg.keepLifecycleScripts) {
+      const npmPublishInfo = isIvy
```
Comment (alan-agius4) [inline src/lib/ng-package/entry-point/write-package.transform.ts:142]: I don't follow the need to a different error message when using Ivy.

Comment (SchnWalter) [reply src/lib/ng-package/entry-point/write-package.transform.ts:142]: I'll revert to using just one message.

--- 

If I remember correctly, this is a legacy from #1711 where after my changes you were allowed to publish Ivy libraries, but after the discussions we dropped some of those changes and I created this MR.

File: package.json [hunk 2]
```diff
@@ -114,6 +114,7 @@
     "postbuild": "node ./scripts/postbuild.js",
     "schema": "json2ts --input src/ng-package.schema.json --output src/ng-package.schema.ts",
     "release": "standard-version --releaseCommitMessageFormat 'release: cut v{{currentTag}}' --no-verify",
+    "postinstall": "yarn schema",
```
Comment (alan-agius4) [inline package.json:117]: Why is this needed?

Comment (SchnWalter) [reply package.json:117]: If you just just run `git clone` and `npm install`, if you open the project inside an IDe, you will get errors in all files that import the schema interface. This fixes that issue.

**Comment by alan-agius4** [comment]
@SchnWalter, are you still interesting in landing this?

**Comment by alan-agius4** [comment]
in that case let's close this.
